### PR TITLE
Do not close singleton instances.

### PR DESCRIPTION
**Pull request by salty-blue-mango**
This must be done by GC when there are no references to it.

### What does this PR do?
Port #50798 to master

### What issues does this PR fix or reference?
#45839

### Tests written?

No

### Commits signed with GPG?

No

### CODE REVIEW

**Comment by xeacott** [comment]
Appears to be some left over output from `diff` in the code. Clean that out and I'm sure this will pass. :)

File: salt/transport/ipc.py [hunk 1]
```diff
@@ -707,12 +701,16 @@ def read_async(self, callback):
...
+    def _close(self):
         '''
         Routines to handle any cleanup before the instance shuts down.
         Sockets and filehandles should be closed explicitly, to prevent
         leaks.
+
+        This class is a singleton so close have to be called only once during
+        garbage collection when nobody uses this instance.
         '''
+<<<<<<< HEAD
```
Comment (dwoz) [inline salt/transport/ipc.py]: @Nate-Watts It looks like there are merge conflicts included in this change

**Review by dwoz — CHANGES_REQUESTED** [review]

**Review by dwoz — CHANGES_REQUESTED** [review]
@Nate-Watts The linter is failing and I am concerned about the amount of test suites that timed out with this change.

**Comment by frogunder** [comment]
@Nate-Watts You mind stopping by this one and look at the lint and test failures?

Thanks.

**Pull request closed without merging**